### PR TITLE
Support --test-xmlrunner=test-reports for junit test report output

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/install/shared_setup.py [hunk 1]
```diff
@@ -304,6 +307,33 @@ def reload_vsc_modules(self):
...
 
+    def force_xmlrunner(self):
+        """
+        A monkey-patch attempt to run the tests with
+        xmlrunner.XMLTestRunner(output=xyz).run(suite)
+
+        E.g. in case of jenkins and you want junit compatible reports
+        """
+        import xmlrunner
+        import unittest
```
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: do we really want these imports here, not on top?

File: lib/vsc/install/shared_setup.py [hunk 2]
```diff
@@ -304,6 +307,33 @@ def reload_vsc_modules(self):
...
+
+        xmlrunner_output = self.test_xmlrunner
+
+        class OutputXMLTestRunner(xmlrunner.XMLTestRunner):
+            """Force the output"""
+            def __init__(self, *args, **kwargs):
+                kwargs['output'] = xmlrunner_output
+                xmlrunner.XMLTestRunner.__init__(self,*args, **kwargs)
+
+        main_orig = unittest.main
```
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: separate with empty line?

also, use this instead?

```python
from unittest import main as unittest_main
```

File: lib/vsc/install/shared_setup.py (hunk 1 above)
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: unittest is ok, but xmlrunner would become a build_requires

File: lib/vsc/install/shared_setup.py (hunk 2 above)
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: no, i don't think you can use that for monley patching

File: lib/vsc/install/shared_setup.py (hunk 1 above)
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: you can just ignore the `ImportError` for `xmlrunner` on top, and complain in here if it's not imported?

File: lib/vsc/install/shared_setup.py (hunk 2 above)
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: `unittest_main` is just a reference to the original `unittest.main`, I don't see a difference with `main_orig` (other than the name ;-))

Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: ok then

Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: but now i need another import line. not doing that :smile:

**Comment by boegel** [comment]
Thanks @stdweird!

**Pull request merged**